src/io/parquet/reader.rs

```diff
@@ -1,89 +1,14 @@
-use parquet::file::metadata::FileMetaData;
 use parquet::file::reader::ChunkReader;
 
-// TODO: deduplicate with `resolve_types` in `downcast.rs`
-fn infer_geo_data_type(
-    geometry_types: &HashSet<&str>,
-    coord_type: CoordType,
-) -> Result<Option<GeoDataType>> {
-    if geometry_types.iter().any(|t| t.contains(" Z")) {
-        return Err(GeoArrowError::General(
-            "3D coordinates not currently supported".to_string(),
-        ));
-    }
-
-    match geometry_types.len() {
-        0 => Ok(None),
-        1 => Ok(Some(match *geometry_types.iter().next().unwrap() {
-            "Point" => GeoDataType::Point(coord_type),
-            "LineString" => GeoDataType::LineString(coord_type),
-            "Polygon" => GeoDataType::Polygon(coord_type),
-            "MultiPoint" => GeoDataType::MultiPoint(coord_type),
-            "MultiLineString" => GeoDataType::MultiLineString(coord_type),
-            "MultiPolygon" => GeoDataType::MultiPolygon(coord_type),
-            "GeometryCollection" => GeoDataType::GeometryCollection(coord_type),
-            _ => unreachable!(),
-        })),
-        2 => {
-            if geometry_types.contains("Point") && geometry_types.contains("MultiPoint") {
-                Ok(Some(GeoDataType::MultiPoint(coord_type)))
-            } else if geometry_types.contains("LineString")
-                && geometry_types.contains("MultiLineString")
-            {
-                Ok(Some(GeoDataType::MultiLineString(coord_type)))
-            } else if geometry_types.contains("Polygon") && geometry_types.contains("MultiPolygon")
-            {
-                Ok(Some(GeoDataType::MultiPolygon(coord_type)))
-            } else {
-                Ok(Some(GeoDataType::Mixed(coord_type)))
-            }
-        }
-        _ => Ok(Some(GeoDataType::Mixed(coord_type))),
-    }
-}
-
-fn parse_geoparquet_metadata(
-    metadata: &FileMetaData,
-    schema: &Schema,
-    coord_type: CoordType,
-) -> Result<(usize, Option<GeoDataType>)> {
-    let meta = GeoParquetMetadata::from_parquet_meta(metadata)?;
-    let column_meta = meta
-        .columns
-        .get(&meta.primary_column)
-        .ok_or(GeoArrowError::General(format!(
-            "Expected {} in GeoParquet column metadata",
-            &meta.primary_column
-        )))?;
-
-    let geometry_column_index = schema
-        .fields()
-        .iter()
-        .position(|field| field.name() == &meta.primary_column)
-        .unwrap();
-    let mut geometry_types = HashSet::with_capacity(column_meta.geometry_types.len());
-    column_meta.geometry_types.iter().for_each(|t| {
-        geometry_types.insert(t.as_str());
-    });
-    Ok((
-        geometry_column_index,
-        infer_geo_data_type(&geometry_types, coord_type)?,
-    ))
-}
-
 pub struct GeoParquetReaderOptions {
-    batch_size: usize,
-    coord_type: CoordType,
+    pub batch_size: usize,
+    pub coord_type: CoordType,
 }
 
 impl GeoParquetReaderOptions {
```

Comment:
Should this `GeoParquetReaderOptions` struct go in a common place like `src/io/parquet/structs.rs`? Since this will be used in both the sync/async GeoParquet readers?